src/rules/jsx_no_danger_with_children.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use super::{Context, LintRule};
+use crate::handler::{Handler, Traverse};
+use crate::Program;
+use deno_ast::view::{JSXAttrName, JSXAttrOrSpread, JSXElement};
+use deno_ast::SourceRanged;
+
+#[derive(Debug)]
+pub struct JSXNoDangerWithChildren;
+
+const CODE: &str = "jsx-no-danger-with-children";
+
+impl LintRule for JSXNoDangerWithChildren {
+  fn tags(&self) -> &'static [&'static str] {
+    &["react", "jsx", "fresh"]
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn lint_program_with_ast_view(
+    &self,
+    context: &mut Context,
+    program: Program,
+  ) {
+    JSXNoDangerWithChildrenHandler.traverse(program, context);
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/jsx_no_danger_with_children.md")
+  }
+}
+
+const MESSAGE: &str =
+  "Using JSX children together with 'dangerouslySetInnerHTML' is invalid";
+const HINT: &str = "Remove the JSX children";
+
+struct JSXNoDangerWithChildrenHandler;
+
+impl Handler for JSXNoDangerWithChildrenHandler {
+  fn jsx_element(&mut self, node: &JSXElement, ctx: &mut Context) {
+    for attr in node.opening.attrs {
+      if let JSXAttrOrSpread::JSXAttr(attr) = attr {
+        if let JSXAttrName::Ident(id) = attr.name {
+          if id.sym() == "dangerouslySetInnerHTML" && !node.children.is_empty()
+          {
+            ctx.add_diagnostic_with_hint(node.range(), CODE, MESSAGE, HINT);
+          }
+        }
+      }
+    }
+  }
+}
+
+// most tests are taken from ESlint, commenting those
+// requiring code path support
```

Comment:
What's code path?

Reply:
Lol I just copied the tests from another existing lint rule we had and looks like I copied that code comment everywhere :joy: